h2o-algos/src/main/java/hex/rulefit/RuleFitModel.java

```diff
@@ -173,4 +177,39 @@ public RuleFitMojoWriter getMojo() {
     public boolean haveMojo() {
         return true;
     }
+    
+    @Override
+    public Frame fitRules(Frame frame, String[] ruleIds) {
+        Frame adaptFrm = new Frame(frame);
+        adaptTestForTrain(adaptFrm, true, false);
+        
+        List<Rule> rules = new ArrayList<>();
+        List<String> linearRules = new ArrayList<>();
+        for (int i = 0; i < ruleIds.length; i++) {
+            if (ruleIds[i].startsWith("linear.") && isLinearVar(ruleIds[i])) {
```

Comment:
Is the first part of the condition necessary?

Reply:
the reason is: if I know that the rule id doesn't start with "linear." then it's not necessary to evaluate isLinearVar() method and loop inside of it. So on easy cases, first part of && is false and it doesn't even evaluate the second part. And most of the cases will be that ruleIds[i] not starting with "linear." and they will fall to else branch right after first part of the condition evaluated.

Reply:
Oh, I see. Sure, make sense. :+1: